search.go

```diff
@@ -402,6 +439,77 @@ func (l *Conn) Search(searchRequest *SearchRequest) (*SearchResult, error) {
 	}
 }
 
+// SearchAsync performs the given search request asynchronously, it returns a SearchAsyncResponse channel which will be
+// closed when the request finished and an error, not nil if the request to the server failed
+func (l *Conn) SearchAsync(ctx context.Context, searchRequest *SearchRequest) (<-chan *SearchAsyncResponse, error) {
```

Comment:
There's currently a discussion in another PR regarding `context.Context` (#406). This would be the only function to accept a context. I suggest to cut out the context handling and wait for the mentioned PR to be merged so we have a streamlined design on how to implement `context.Context`.

Reply:
Sorry about that. I'll remove it, but we need a way to stop the receive loop.
Maybe take an optional `done` channel as parameter to stop when it is closed, e.g.:
```go
// SearchAsync performs the given search request asynchronously, it takes an optional done channel to stop the request. It returns a SearchAsyncResponse channel which will be
// closed when the request finished and an error, not nil if the request to the server failed
func (l *Conn) SearchAsync(searchRequest *SearchRequest, done chan struct{}) (<-chan *SearchAsyncResponse, error)
```